glitter/pages/admin.py

```diff
@@ -71,5 +77,64 @@ def wrapper(*args, **kwargs):
         urlpatterns = [
             url(r'^$', wrap(self.grid_view)),
             url(r'^tree/$', wrap(self.changelist_view), name='%s_%s_tree' % info),
+            url(r'^(\d+)/duplicate/$', wrap(self.duplicate_page), name='%s_%s_duplicate' % info),
         ] + urlpatterns
         return urlpatterns
+
+    def duplicate_page(self, request, obj_id):
+        obj = get_object_or_404(Page, id=obj_id)
+
+        if request.method == "POST":
+            form = DuplicatePageForm(request.POST or None)
+            if form.is_valid():
+                new_page = form.save()
+
+                # Use current version if exists if not get the latest
+                if obj.current_version:
```

Comment:
A page might not have a `Version` associated with it at all, and it currently errors out.